packages/fx-core/src/plugins/resource/frontend/index.ts

```diff
@@ -44,20 +44,23 @@ export class FrontendPlugin implements Plugin {
   dotnetPluginImpl = new DotnetPluginImpl();
 
   private static setContext(ctx: PluginContext): void {
-    Logger.setLogger(ctx.logProvider);
-    TelemetryHelper.setContext(
-      ctx,
-      isVsCallingCli() ? DotnetPluginInfo.pluginName : FrontendPluginInfo.PluginName
-    );
+    const component = this.isVsPlatform(ctx)
```

Comment:
Is it better to get implement instance from context as below, so that we don't need to use `this.isVsPlatform(ctx)?...` across many method such as scaffold/provision/deploy etc.

```
private static getImpl(ctx: PluginContext): ImplInterface {
  const impl = this.isVsPlatform(ctx) ? this.dotnetPluginImpl : this.frontendPluginImpl;
  Logger.setLogger(ctx.logProvider, impl.pluginName);
  TelemetryHelper.setContext(ctx, impl.pluginName);
  return impl;
}

public async scaffold(ctx: PluginContext): Promise<TeamsFxResult> {
  const impl = FrontendPlugin.getImpl(ctx);
  return this.runWithErrorHandling(ctx, TelemetryEvent.Scaffold, () => impl.scaffold(ctx));
```
ImplInterface (can be any more proper name) is the interface of frontendPluginImpl and dotnetPluginImpl.

Reply:
Updated.